Ebb fixes for fpic and ppc32 handler

pmur opened the pull request:
This started as an attempt to fix the -fpic issue. I uncovered several issues with the ppc32 ebb handlers when I noticed the backtrace test was failing under ppc32. This should address issues #10 and #11.

This should resolve them. The PPC64 handler will likely need some similar fixes (Save/restore VSCR, and saving 512B stack in the VR handler).

pmur commented:
More updates. I've greatly overhauled the clobber test, and simplified the PPC32 vector load/stores based on the ABI, and fixed a handful of bugs in the PPC64 handler.

pmur commented:
Another update. I removed redundant ASM directives in the updated test as only P8 supports EBB. I've also unified the ppc32/ppc64 handlers into one file. It simplifies things greatly as they are verify similar.

gftg85 commented on ebb/ebb-callback-ppc32.S:
```diff
@@ -36,135 +36,202 @@
- *   r1 + STACK_OFFSET+4         |  LR
- *   r1 + STACK_OFFSET           |  CR
+ *   r1 + 152:160   |  Alignment padding
+ *   r1 + 32:152    |  GRP r0,r3-r31
```
Is this a typo? GPR instead of GRP?

gftg85 commented on ebb/ebb-callback-ppc32.S:
```diff
@@ -36,135 +36,202 @@
+ *
+ *   --------------  __paf_ebb_callback_handler_fpr_[tcb|tls]  -----------------
+ *   r1 + 408       |  FPSCR (64b store)
+ *   r1 + 152:408   |  FRP f0:f31
```
FPR instead of FRP?

tuliom commented on ebb/ebb-callback.S:
```diff
@@ -0,0 +1,613 @@
+/* Event-Based Branch Facility API.  PowerPC32 callback handler.
+ *
+ * Copyright IBM Corp. 2013,2015
```
I guess this should be 2013-2015.

tuliom commented on ebb/ebb-callback.S:
```diff
@@ -0,0 +1,613 @@
+/* Event-Based Branch Facility API.  PowerPC32 callback handler.
```
Is this really PowerPC32 only?

pmur replied:
It covers all three ABI's now. That comment should be updated. I've tested and verified the generated code on all three.

tuliom commented on ebb/tests/test_ebb_backtrace.c:
```diff
@@ -1,6 +1,6 @@
 /* Event-Based Branch Facility Tests.
  *
- * Copyright IBM Corp. 2013
+ * Copyright IBM Corp. 2013,2015
```
Same thing: 2013-2015

tuliom replied:
This copyright entry still needs to be fixed.

tuliom commented on ebb/tests/test_ebb_pmu_clobber_hooks.S:
```diff
@@ -0,0 +1,514 @@
+	EPILOGUE(vsr)
+END(ebb_test_vsr)
+
+#if 0
```
What is this doing here?

pmur commented:
I figured out I wasn't properly setting up the TOC for elfv2 and implemented the clobber functions which had been previously commented out. They now clobber all the common ABI volatile registers.

Otherwise, I addressed the minor comment comments.

tuliom commented:
With that small copyright fix, this will be good to me.

pmur commented:
Rebased onto master with the requested copyright formatting fix to ebb/test_ebb_backtrace.c.

Thanks all for the review!
